lib/src/compile.ts

```diff
@@ -87,9 +89,17 @@ function newCompileStringRequest(
   input.setSource(source);
   input.setSyntax(utils.protofySyntax(options?.syntax ?? 'scss'));
 
-  if (options?.url) input.setUrl(options.url.toString());
+  const url = options?.url?.toString();
+  if (url && url !== legacyImporterProtocol) {
+    input.setUrl(url);
+  }
 
-  if (options && 'importer' in options && options.importer) {
+  const hasImporter = options && 'importer' in options && options.importer;
+  if (url && !hasImporter) {
+    const importer = new proto.InboundMessage.CompileRequest.Importer();
+    importer.setPath(p.resolve('.'));
+    input.setImporter(importer);
+  } else if (hasImporter) {
     input.setImporter(importers.register(options.importer));
   }
```

Comment:
As of https://github.com/sass/dart-sass-embedded/pull/83, not setting importer would create a `noOp` importer on compiler side instead of `FilesystemImporter('.')`, thus it would fix https://github.com/sass/dart-sass-embedded/issues/82.